d4/src/task/perc_cov.rs

```diff
@@ -0,0 +1,77 @@
+use super::{Task, TaskPartition};
+use std::iter::Once;
+
+#[derive(Clone)]
+pub struct PercentCovPart {
+    //thresholds should be sorted
+    thresholds: Vec<u32>,
+    counts: Vec<u32>,
+}
+
+impl TaskPartition<Once<i32>> for PercentCovPart {
+    type ParentType = PercentCov;
+    type ResultType = Vec<u32>;
+    fn new(left: u32, right: u32, parent: &Self::ParentType) -> Self {
+        Self {
+            thresholds: parent.3.clone(),
+            counts: vec![0; parent.3.len()],
+        }
+    }
+    // #[inline(always)]
+    fn feed(&mut self, _: u32, value: &mut Once<i32>) -> bool {
+        let value = value.next().unwrap();
+        for (i, thresh) in self.thresholds.iter().enumerate() {
+            if value as u32 >= *thresh {
+                self.counts[i] += 1
+            }
+        }
+        true
+    }
+    // #[inline(always)]
+    fn feed_range(&mut self, left: u32, right: u32, value: &mut Once<i32>) -> bool {
+        let value = value.next().unwrap();
+        for (i, thresh) in self.thresholds.iter().enumerate() {
+            if value as u32 >= *thresh {
+                self.counts[i] += right - left as u32
+            }
```

Comment:
since thresholds are sorted, you can break in else, but that might be a net negative in most common case where there are few thresholds

Reply:
ah that makes sense to me. will add.

Reply:
changed my mind on this, i think you're right that the most common case will be few thresholds